src/EventListener/MapToListener.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AutoMapper\EventListener;
+
+use AutoMapper\Attribute\MapTo;
+use AutoMapper\Event\GenerateMapperEvent;
+use AutoMapper\Event\PropertyMetadataEvent;
+use AutoMapper\Event\SourcePropertyMetadata;
+use AutoMapper\Event\TargetPropertyMetadata;
+use AutoMapper\Exception\BadMapDefinitionException;
+use AutoMapper\Transformer\CallableTransformer;
+use AutoMapper\Transformer\CustomTransformer\CustomModelTransformer;
+use AutoMapper\Transformer\CustomTransformer\CustomModelTransformerInterface;
+use AutoMapper\Transformer\CustomTransformer\CustomPropertyTransformer;
+use AutoMapper\Transformer\CustomTransformer\CustomPropertyTransformerInterface;
+use AutoMapper\Transformer\CustomTransformer\CustomTransformersRegistry;
+
+final readonly class MapToListener
+{
+    public function __construct(private CustomTransformersRegistry $customTransformersRegistry)
+    {
+    }
+
+    public function __invoke(GenerateMapperEvent $event): void
+    {
+        if ($event->mapperMetadata->sourceReflectionClass === null) {
+            return;
+        }
+
+        $properties = $event->mapperMetadata->sourceReflectionClass->getProperties();
+
+        foreach ($properties as $reflectionProperty) {
+            $mapToAttributes = $reflectionProperty->getAttributes(MapTo::class);
+
+            if (empty($mapToAttributes)) {
+                continue;
+            }
+
+            foreach ($mapToAttributes as $mapToAttribute) {
```

Comment:
Do we have some kind of priority between attributes there ? Or it's first written onto the class that is first here ?

Reply:
No reason to have priority since we throw an exception when we have a rule for the same target.